wkcuber/image_readers.py

```diff
@@ -150,34 +154,54 @@ def find_count_of_axis(tif_file: TiffFile, axis: str) -> int:
 
     def read_array(self, file_name: str, dtype: np.dtype, z_slice: int) -> np.ndarray:
         with TiffFile(file_name) as tif_file:
-            num_channels = self.read_channel_count(file_name)
-            if len(tif_file.pages) > num_channels:
+            if self.num_channels is None:
+                self.num_channels = self.read_channel_count(file_name)
+            if self.is_page_multi_channel is None:
+                # we assume all tif pages have the same dimensions
+                self.is_page_multi_channel = tif_file.pages[0].ndim == 3
+
+            if self.is_page_multi_channel:
+                self.num_channels = 1
+
+            if len(tif_file.pages) > self.num_channels:
                 data = np.array(
                     list(
                         map(
                             lambda x: x.asarray(),
                             tif_file.pages[
-                                z_slice * num_channels : z_slice * num_channels
-                                + num_channels
+                                z_slice
+                                * self.num_channels : z_slice
+                                * self.num_channels
+                                + self.num_channels
                             ],
                         )
                     ),
                     dtype,
                 )
             else:
                 data = np.array(
-                    list(map(lambda x: x.asarray(), tif_file.pages[0:num_channels])),
+                    list(
+                        map(
+                            lambda x: x.asarray(), tif_file.pages[0 : self.num_channels]
+                        )
+                    ),
                     dtype,
                 )
+
+            # if the pages are multi-channel, then we'll have 4 dimensions here because of [x:x+1] notation, so we reshape the data
+            if self.is_page_multi_channel:
+                data = data.reshape((data.shape[1], data.shape[2], data.shape[3]))
+                x_index = tif_file.pages[0].axes.find("X")
+                y_index = tif_file.pages[0].axes.find("Y")
+                c_index = tif_file.pages[0].axes.find("S")
```

Comment:
Can the axis also be `C` (instead of `S`)? 
Consider renaming the variable to `s_index`.

Reply:
All example files always used the `S` index if a page was multi-channel. That's why I assume it here. To use the same variable in both if-branches, I would leave as `c_index`, if that's fine for you

Reply:
Ok, maybe I misunderstood it. If the axis are `XYC`, this code still works even though you are not checking for `C`, right?

Reply:
Right because as far as I know if `self.is_page_multi_channel` is `true`, there is no `C`